pkg/controller/user-cluster-controller-manager/application-installation-controller/controller.go

```diff
@@ -214,6 +218,20 @@ func (r *reconciler) getApplicationVersion(appInstallation *appskubermaticv1.App
 	return fmt.Errorf("application version '%s' does not exist in applicationDefinition %s", desiredVersion, applicationDef.Name)
 }
 
+func setAppNamespace(ctx context.Context, seedClient ctrlruntimeclient.Client, applicationInstallation *appskubermaticv1.ApplicationInstallation) error {
+	applicationDefinition := appskubermaticv1.ApplicationDefinition{}
+	if err := seedClient.Get(ctx, types.NamespacedName{Name: applicationInstallation.Spec.ApplicationRef.Name}, &applicationDefinition); err != nil {
+		return fmt.Errorf("error on fetching application definition for setting appinstallation namespace. %w", err)
```

Comment:
`: %w` at the end for better error formatting. Also you can get rid of "for setting appinstallation namespace", because the calling function will prepend a very similar text to the error message already.